middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -245,9 +264,13 @@ let unary_int_arith_primitive _env dbg kind op arg =
     C.bswap16 arg dbg
   (* Special case for manipulating int64 on 32-bit hosts *)
   | Naked_int64, Neg when Target_system.is_32_bit -> C.unsupported_32_bit ()
+  (* Negation needs a sign-extension for 32-bit and 63-bit values *)
+  | Naked_immediate, Neg ->
+    sign_extend_63 dbg (C.sub_int (C.int ~dbg 0) arg dbg)
```

Comment:
here, we could probably apply `low_63` to arg, no ?

Reply:
Yeah I think this is correct, I've added it.